### ISSUE TITLE
Export trained model to another notebook.

### ISSUE THREAD

**Issue by meetgandhi123** [facebookresearch/detectron2#2081]
How to export a custom model to another google collab notebook for prediction on custom images. 

I used [link](https://colab.research.google.com/drive/16jcaJoc6bCFAQ96jDe2HwtXj7BMD_-m5) detectron2 tutorial google collab notebook and performed all the operations for "Train on a custom dataset" which gave me a custom model for detecting balloon. And the model was also running on custom images in that notebook.

In order to save the model I used the following code and saved the output folder in my google drive:

```
# Save config to file
with open("output/config.yaml", "w") as f:
    f.write(cfg.dump())

# Saving the output folder
import shutil
shutil.make_archive('balloon_model', 'zip', '/content/output')
%cp "/content/balloon_model.zip" "/content/drive/My Drive/" 
```

In the new notebook, for predicting the custom images I used the following code:

```
#importing output folder
%mkdir output
%cd output
!unzip "/content/drive/My Drive/balloon_model.zip"
%cd ..

#setting up the model
import os
cfg = get_cfg()
cfg.MODEL.WEIGHTS = os.path.join(cfg.OUTPUT_DIR, "model_final.pth")
cfg.DATASETS.TEST = ("my_dataset_test", )
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.7   # set the testing threshold for this model
predictor = DefaultPredictor(cfg)
test_metadata = MetadataCatalog.get("my_dataset_test")

```

Output:

```
Skip loading parameter 'proposal_generator.rpn_head.conv.weight' to the model due to incompatible shapes: (256, 256, 3, 3) in the checkpoint but (1024, 1024, 3, 3) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.rpn_head.conv.bias' to the model due to incompatible shapes: (256,) in the checkpoint but (1024,) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.rpn_head.objectness_logits.weight' to the model due to incompatible shapes: (3, 256, 1, 1) in the checkpoint but (15, 1024, 1, 1) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.rpn_head.objectness_logits.bias' to the model due to incompatible shapes: (3,) in the checkpoint but (15,) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.rpn_head.anchor_deltas.weight' to the model due to incompatible shapes: (12, 256, 1, 1) in the checkpoint but (60, 1024, 1, 1) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.rpn_head.anchor_deltas.bias' to the model due to incompatible shapes: (12,) in the checkpoint but (60,) in the model! You might want to double check if this is expected.
Skip loading parameter 'proposal_generator.anchor_generator.cell_anchors.0' to the model due to incompatible shapes: (3, 4) in the checkpoint but (15, 4) in the model! You might want to double check if this is expected.
Skip loading parameter 'roi_heads.box_predictor.cls_score.weight' to the model due to incompatible shapes: (2, 1024) in the checkpoint but (81, 2048) in the model! You might want to double check if this is expected.
Skip loading parameter 'roi_heads.box_predictor.cls_score.bias' to the model due to incompatible shapes: (2,) in the checkpoint but (81,) in the model! You might want to double check if this is expected.
Skip loading parameter 'roi_heads.box_predictor.bbox_pred.weight' to the model due to incompatible shapes: (4, 1024) in the checkpoint but (320, 2048) in the model! You might want to double check if this is expected.
Skip loading parameter 'roi_heads.box_predictor.bbox_pred.bias' to the model due to incompatible shapes: (4,) in the checkpoint but (320,) in the model! You might want to double check if this is expected.
```

```
#testing the images:
from detectron2.utils.visualizer import ColorMode
import glob

im = cv2.imread("/content/download (2).jpg")
outputs = predictor(im)
v = Visualizer(im[:, :, ::-1],
              metadata=test_metadata, 
              scale=0.8
                )
out = v.draw_instance_predictions(outputs["instances"].to("cpu"))
cv2_imshow(out.get_image()[:, :, ::-1])

```

But the input image and the output image is the same. 
![download (2)](https://user-images.githubusercontent.com/28274170/94547298-b917cb80-026c-11eb-9718-f6f07f16e47a.jpg)

**Comment by ppwwyyxx** [comment]
According to the provided code you did not load the config that you saved.

**Comment by meetgandhi123** [comment]
> According to the provided code you did not load the config that you saved.

If possible ,can you please tell me how to do that. It would be a great help..

**Comment by ppwwyyxx** [comment]
The documentation of metadata https://detectron2.readthedocs.io/tutorials/datasets.html#metadata-for-datasets has explained how to set labels.

**Comment by meetgandhi123** [comment]
Thankyou very much for your support.

**Comment by AbhishekMaity001** [comment]
Solution for this problem is simple... though it took me hours to figure it out .... 👍 
from detectron2.data import MetadataCatalog
MetadataCatalog.get("your_dataset_name_you_registered").thing_classes = ['class1','class2','class3'] # same no. cls tht u have
v = Visualizer(image[:, :, ::-1],
               scale=4,
               metadata=MetadataCatalog.get("your_dataset_name_you_registered"), # include this line mainly
               instance_mode=ColorMode.IMAGE
               )
